WordPress/Classes/Networking/ThemeServiceRemote.m

```diff
@@ -118,16 +111,53 @@ - (NSOperation *)getThemesForBlogId:(NSNumber *)blogId
 {
     NSParameterAssert([blogId isKindOfClass:[NSNumber class]]);
 
+    NSMutableArray *themes = [NSMutableArray array];
     NSString *path = [NSString stringWithFormat:@"sites/%@/themes", blogId];
+    
+    NSOperation *operation = [self getThemesPage:1
+                                          themes:themes
+                                            path:path
+                                         success:success
+                                         failure:failure];
+    
+    return operation;
+}
+
+- (NSOperation *)getThemesPage:(NSInteger)page
+                        themes:(NSMutableArray *)themes
+                          path:(NSString *)path
+                       success:(ThemeServiceRemoteThemesRequestSuccessBlock)success
+                       failure:(ThemeServiceRemoteFailureBlock)failure
+{
     NSString *requestUrl = [self pathForEndpoint:path
-                                     withVersion:ServiceRemoteRESTApiVersion_1_1];
+                                     withVersion:ServiceRemoteRESTApiVersion_1_2];
+    
+    static NSString* const ThemeRequestTierKey = @"tier";
+    static NSString* const ThemeRequestTierAllValue = @"all";
+    static NSString* const ThemeRequestNumberKey = @"number";
+    static NSInteger const ThemeRequestNumberValue = 100;
+    static NSString* const ThemeRequestPageKey = @"page";
 
+    NSDictionary *parameters = @{ThemeRequestTierKey: ThemeRequestTierAllValue,
+                                 ThemeRequestNumberKey: @(ThemeRequestNumberValue),
+                                 ThemeRequestPageKey: @(page),
+                                 };
+    
     NSOperation *operation = [self.api GET:requestUrl
-                                parameters:nil
+                                parameters:parameters
                                    success:^(AFHTTPRequestOperation *operation, NSDictionary *response) {
-                                       if (success) {
-                                           NSArray *themes = [self themesFromMultipleThemesRequestResponse:response];
-                                           success(themes);
+                                       NSArray *pageThemes = [self themesFromMultipleThemesRequestResponse:response];
+                                       [themes addObjectsFromArray:pageThemes];
+                                       NSInteger themesCount = [[response numberForKey:ThemeServiceRemoteThemeCountKey] integerValue];
+                                       if (pageThemes.count && themes.count < themesCount) {
+                                           NSInteger nextPage = page + 1;
+                                           [self getThemesPage:nextPage
```

Comment:
:+1: i'd like to suggest that (load on scroll), otherwise the user just needs to wait untill _all of the themes_ were fetched. And that might be a very bad thing, depending on the connection speed.